### PR TITLE
MLv2 new expression syntax

### PR DESCRIPTION
**Pull request by metamben**
This PR makes `filter`, `aggregate`, and `expression` accept clauses in the forms (JS)
```js
{operator: "=", args: [someField, someNumber, ...]}
```
and (CLJC)
```clojure
{:operator :=, :args [some-field some-number ...]}
{:operator "=", :args [some-field some-number ...]}
```

`current-filter`, `current-filters`, and `resolve-expression` return filters in this form with a string value as `:operator`. (Aggregations are not currently returned at all.)
The closures defined by `defop` return clauses in this form too.

### CODE REVIEW

File: src/metabase/lib/dispatch.cljc [hunk 1]
```diff
@@ -21,5 +21,10 @@
   ;; of [[toucan2.core/model]]?
   (or (mbql-clause-type x)
       (when (map? x)
-        (:lib/type x))
+        (if (and (or (keyword? (:operator x))
+                     (string? (:operator x)))
+                 (vector? (:args x))
+                 (map? (:options x {})))
+          :lib/external-op
+          (:lib/type x)))
```
[the remark this answers is not in the document]
Comment (metamben) [reply src/metabase/lib/dispatch.cljc:29]: I had such a version at one point but didn't like the idea of having to add that info at every entry point. I was also unsure about how this should/would look like at the exit point. Will the namespace be stripped from `lib/type` or should I remove the key altogether?
@camsaul Anyway, if you find adding that property better, I'll do the change.

[the remark this answers is not in the document]
Comment (camsaul) [reply src/metabase/lib/dispatch.cljc:29]: Namespaces shouldn't get stripped out at any point, if they do it's a bug

[the remark this answers is not in the document]
Comment (camsaul) [reply src/metabase/lib/dispatch.cljc:29]: On another note, I think it makes sense to have `lib.convert` or Malli coercion stuff take care of converting `:operator` to a keyword. To get a map with `:operator` as a key you're going to have to do some `js->clj` conversion anyway. I don't think we want to have to make the entire lib JS aware and able to handle conversion, that should all happen at the JS/Cljs boundaries

**Review by camsaul — APPROVED** [review]
works for me

I don't really love having yet another alternate MBQL syntax, but if the frontend has to be constructing MBQL expressions directly, this is better than having it try to destructure and build vectors itself

**Comment by metamben** [comment]
> I don't really love having yet another alternate MBQL syntax, but if the frontend has to be constructing MBQL expressions directly, this is better than having it try to destructure and build vectors itself

I merge this now and will get back to it on my return.

**Pull request merged**